c10/core/Contiguity.h

```diff
@@ -13,15 +13,15 @@ namespace c10 {
 template <typename T>
 bool _compute_contiguous(ArrayRef<T> sizes, ArrayRef<T> strides, T numel) {
   bool is_contiguous = true;
-  if (TORCH_GUARD_SIZE_OBLIVIOUS(sym_eq(numel, 0))) {
+  if (TORCH_GUARD_OR_FALSE(sym_eq(numel, 0))) {
     return is_contiguous;
   }
   T z = 1;
   // NB: make sure we do signed arithmetic
   for (int64_t d = int64_t(sizes.size()) - 1; d >= 0; d--) {
     const auto& size_d = sizes[d];
-    if (TORCH_GUARD_SIZE_OBLIVIOUS(sym_ne(size_d, 1))) {
-      if (TORCH_GUARD_SIZE_OBLIVIOUS(sym_eq(strides[d], z))) {
+    if (TORCH_GUARD_OR_TRUE(sym_ne(size_d, 1))) {
+      if (TORCH_GUARD_OR_FALSE(sym_eq(strides[d], z))) {
```

Comment:
this one is the only tricky one here, we can match preview  guard_size_oblivious trivially if we do, 
```
fun(z, strides[d])
   if knonw_zero_or_one(z) or knonw_zero_or_one(strides[d]):
       return guard_or_false(sym_eq(strides[d], z))
   else:
     return sym_eq(strides[d], z)
```
the other choice is to keep it the way you made it but would need more consults in that case. 

<img width="1173" alt="Screenshot 2025-04-01 at 11 02 34 PM" src="https://github.com/user-attachments/assets/b3b185fd-a7f9-4151-92fc-5b2ea8c266f8" />

In general I feel the long term right fix here is 
to have two functions
_compute_contiguous and
_maybe_compute_contiguous or _known_contiguous
where the latter is called when its ok to return false for True